### PR TITLE
The basics of ftl movement

### PR DESCRIPTION
**Pull request by TMTIME**
~~thats it my work is done~~

srs tho posting this now so if @Qustinnus wants to merge so he can get his stuff working with it he can

FTL is fast as fuck for testing. Will add more timers later.

Adds the basics of generating sectors.

maps are now both an empty z level with /turf/open/space/transit/ftl turfs + a spawn landmark
The spawn landmark spawns in the actual ship, which is stored as a template

Hopefully has unit tests to check if ship templates don't use noop turfs where it should

Unvisited systems are qdeled.

throwforce and knockdown is kept bc what a meme

fuck tgui

Unique named sectors that spawn only once per round are supported.

### CODE REVIEW

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+/datum/controller/subsystem/ftl_navigation/Initialize(timeofday)
+	for(var/A in get_areas(/area/, TRUE)) //Get all areas owned by the ship
+		var/area/place = A
+		if(place.z != 2) continue
+		ship_areas += place
+
+	for(var/type in subtypesof(/datum/planet)) //Get all planet types and weights
+		var/datum/planet/P = new type
+		planet_types += type
+		planet_types[type] = P.frequency
```
Comment (Qustinnus) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Also last year ninja discovered you dont have to init these. Remove the first line that makes the instance and change p.frequency to initial[type.frequency] do the same for the others

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	visited_sectors += to_sector
+	to_sector = null
+	for(var/datum/sector/S in current_sector.connected_sectors)
+		generate_connecting_sectors(S)
+	ftl_state = FTL_IDLE
+	message_admins("FTL has ended")
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_parallax(var/ftl_start = TRUE)
+	for(var/area/A in ship_areas)
```
Comment (Qustinnus) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: This one can also be typeless

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+/datum/controller/subsystem/ftl_navigation/proc/ftl_handle_transit_turfs(var/ftl_start = TRUE)
+	var/throw_dir
+	var/flavor_text
+	if(ftl_start)
+		throw_dir = WEST
+		flavor_text = "<span class='notice'>You feel the ship lurch as it enters FTL.</span>"
+	else
+		throw_dir = EAST
+		flavor_text = "<span class='notice'>You feel the ship lurch as it exits FTL.</span>"
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION))
```
Comment (Qustinnus) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: How performant is this action because you're looping through the contents of the entire ma

Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: Also why don't you use ship_areas for this

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+							var/range = jump_throw_force
+							var/speed = range/5
+							M.throw_at(target,range,speed)
+						if(jump_knockdown_force)
+							M.Knockdown(jump_knockdown_force)
+
+
+// /datum/controller/subsystem/starmap/proc/ftl_message(var/message)
+// 	for(var/obj/machinery/computer/ftl_navigation/C in ftl_consoles)
+// 		C.status_update(message)
```
Comment (Qustinnus) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: If unused or not planning to use you should remove this

File: ftl/code/datums/sector.dm [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    station = new /datum/station
+
+/datum/sector/named/dolos
+  name = "Dolos 1" //The intention is to let an antag emag the console to force a jump to this system.
+  frequency = -1 //Will not randomly spawn, but is unique
+  sector_type = "Syndicate Homeworld"
+
+/datum/sector/named/nt_home
+  name = "Gooftown" //The intention is to let an antag emag the console to force a jump to this system.
+  frequency = -1 //Will not randomly spawn, but is unique
```
Comment (Qustinnus) [inline ftl/code/datums/sector.dm]: -1 might fuck with pick weight calc

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+// /datum/controller/subsystem/ftl_navigation/fire()
+// 	ddd
+
+
+// /datum/controller/subsystem/ftl_navigation/Recover()
+// 	ddd
+
+
+/datum/controller/subsystem/ftl_navigation/proc/get_sector(sectorid)
+	for(var/datum/sector/S in current_sector.connected_sectors)
```
[the remark this answers is not in the document]
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: ~~i just stole shuttle id code don't judge~~

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_init_spoolup(sectorid)
+	if(ftl_state != FTL_IDLE) return
+	ftl_state = FTL_SPOOLUP //Setting early to prevent multiple spoolups
+	var/datum/sector/S = get_sector(sectorid)
+	if(!S) return
+	to_sector = S
+	addtimer(CALLBACK(src, .proc/qdel_old_systems, current_sector, to_sector), 20)
```
[the remark this answers is not in the document]
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: proc name is bad.

tldr we only delete systems the crew never visited for now

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 3 above)
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: I need to call begin_ftl() on every transit turf to change how its displayed for people without parallax (not finished this yet)

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 4 above)
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: I plan to use this later

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 3 above)
Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: right, also im guessing planets arent functional? TG has lavaplanet in parallax since I added it but you might want to look into changing that to our own model (probably in a seperate pR)

Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: Also re performance, this took 0.2 seconds on Delta station

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 6 above)
[the remark this answers is not in the document]
Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: its not really bad and I'm not sure if my method is much better or is even viable

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 7 above)
[the remark this answers is not in the document]
Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: ah okay

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 3 above)
Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: did you use the profiler for that or are you saying it froze 0.2 seconds :^)

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 4 above)
Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: ok cool

**Comment by TMTIME** [comment]
Done most of what was asked bar stuff that needs Ninja's input

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 3 above)
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: world.timeofday dif

File: ftl/code/datums/sector.dm (hunk 5 above)
Comment (TMTIME) [reply ftl/code/datums/sector.dm]: now only add systems to the weighted list if they have a valid frequency

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 8]
```diff
@@ -35,79 +31,55 @@ SUBSYSTEM_DEF(ftl_navigation)
...
-	// if(count > 0)
-	// 	msg = "Psst hey. I just lagged the server for [(world.timeofday - timer)/10] seconds due to [count] UHH BASETURF?? on the ship z level. Use /turf/open/space/transit/ftl instead"
-	// 	to_chat(world, "<span class='boldannounce'>[msg]</span>")
-	// 	warning(msg)
-
-	return ..()
-
-
 
+	//baseturf code when floyd is done with SpaceManiac goes here???
```
Comment (Qustinnus) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: im sucking him really hard atm

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+	var/list/planet_types = list()
+
+	var/jump_knockdown_force = 3
+	var/jump_throw_force = 0 //Badmin var
+
+
+/datum/controller/subsystem/ftl_navigation/Initialize(timeofday)
+	for(var/A in get_areas(/area/, TRUE)) //Get all areas owned by the ship
+		var/area/place = A
+		if(place.z != 2) continue
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Don't use single line control statements like this
Also `SSmapping.areas_in_z`

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+		ship_areas += place
+
+	for(var/type in subtypesof(/datum/planet)) //Get all planet types and weights
+		var/datum/planet/P = type
+		planet_types += P
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: This line is unnecessary

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+		planet_types[type] = initial(P.frequency)
+
+	for(var/type in subtypesof(/datum/sector)) //Get all sector types and weights
+		var/datum/sector/S = type
+		if(initial(S.frequency) > 0)
+			sector_types += S
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Same as above

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+			sector_types[type] = initial(S.frequency)
+
+
+	current_sector = new /datum/sector/named/nt_home //Only give them a basic sector to start in
+	visited_sectors += current_sector
+	generate_connecting_sectors(current_sector)
+	for(var/s in current_sector.connected_sectors)
+		var/datum/sector/S = s
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Please no. Press more than one key

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+		generate_connecting_sectors(S)
+
+	var/timer = world.timeofday
+	var/count = 0
+	var/msg
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION))
+		for(var/datum/sub_turf_block/STB in split_block(locate(1,1,z), locate(255,255,z)))
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: What's the point of this sub_turf_block?

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+			for(var/t in STB.return_list())
+				var/turf/T = t
+				if(T.type == /turf/open/space || T.type == /turf/open/space/basic) //I want types only, not subtypes too
+					T.ChangeTurf(/turf/open/space/transit/ftl)
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: As I said before you should load the ship as a template onto a blank z level of transit space in the first place. Then you just have noop in the template

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+		msg = "Psst hey. I just lagged the server for [(world.timeofday - timer)/10] seconds due to [count] wrong space tiles on the ship z level. Use /turf/open/space/transit/ftl instead"
+		to_chat(world, "<span class='boldannounce'>[msg]</span>")
+		warning(msg)
+
+	//baseturf code when floyd is done with SpaceManiac goes here???
+
+	return ..()
+
+
+/datum/controller/subsystem/ftl_navigation/proc/get_sector(sectorid)
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Just have a list of all sectors in an `id:sector_datum` format

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+	for(var/s in current_sector.connected_sectors)
+		var/datum/sector/S = s
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: PLEASE

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+		if(S.id == sectorid) //Try sectors we can jump to
+			return S
+		for(var/s2 in S.connected_sectors)
+			var/datum/sector/S2 = s2
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: I BEG OF YOU

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+			if(S2.id == sectorid) //Try those sectors
+				return S2
+	for(var/s3 in visited_sectors)
+		var/datum/sector/S3 = s3
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: NO MORE

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+		if(S3.id == sectorid) //As a last ditch, try sectors we ever visited.
+			return S3
+	ftl_state = FTL_IDLE
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: The hell? Why are you setting ftl_state in a proc to get a sector by id?

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+	WARNING("No sector ID found. Stopped spoolups/jumps just incase")
+
+
+
+
+/datum/controller/subsystem/ftl_navigation/proc/ftl_init_spoolup(sectorid)
+	if(ftl_state != FTL_IDLE) return
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Dont use single line control statements

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 21]
```diff
@@ -0,0 +1,198 @@
...
+	ftl_state = FTL_SPOOLUP //Setting early to prevent multiple spoolups
+	var/datum/sector/S = get_sector(sectorid)
+	if(!S) return
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Same as above

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+	current_sector = null
+	spool_up() //Begin FTL
+	return TRUE
+
+
+/datum/controller/subsystem/ftl_navigation/proc/qdel_unvisited_systems(var/datum/sector/oldS, var/datum/sector/toS)
+	for(var/s in oldS.connected_sectors) //Cleans up unvisited sectors.
+		var/datum/sector/S = s
+		oldS.connected_sectors -= S
+		if(S != toS) qdel(S)
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Same as above

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 23]
```diff
@@ -0,0 +1,198 @@
...
+
+
+/datum/controller/subsystem/ftl_navigation/proc/generate_connecting_sectors(var/datum/sector/S)
+	var/datum/sector/N
+	var/new_sector
+	for(var/a in 1 to rand(1,3))
+		new_sector = pickweight(sector_types)
+		N = new new_sector
+		if(N.unique) sector_types -= new_sector //If the sector is unique remove it from the spawn pool
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Same as above

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 24]
```diff
@@ -0,0 +1,198 @@
...
+		for(var/datum/sub_turf_block/STB in split_block(locate(1,1,z), locate(255,255,z)))
+			for(var/t in STB.return_list())
+				var/turf/T = t
+				if(istype(T,/turf/open/space/transit/ftl))
+					var/turf/open/space/transit/ftl/F = T
+					F.ftl_turf_update(ftl_start)
+				for(var/obj/machinery/light/L in T.contents) //Makes lights flicker, since I liked how lighting would falter during FTL
+					L.flicker (4,15)
+				for(var/mob/living/M in T.contents) //Messing with players
+					if(M.buckled && M.client) shake_camera(M,3,1)
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: Same as above

File: ftl/code/machinery/computer/navigation_computer.dm [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+	. = ..()
+	if(mode != "Spoolup" && SSftl_navigation.ftl_state == FTL_SPOOLUP) mode = "Spoolup" //Fuck popup uis
+	if(mode != "Transit" && SSftl_navigation.ftl_state == FTL_JUMPING) mode = "Transit"
+	if((mode == "Spoolup" || mode == "Transit") && SSftl_navigation.ftl_state == FTL_IDLE) mode = "Options"
+	var/dat = "fuck tgui: MODE | [mode]<br><br>"
+	switch(mode)
+		if("Options")
+			dat += "Currently located in the [SSftl_navigation.current_sector.name] sector.<br>This is \a [SSftl_navigation.current_sector.sector_type].<br>"
+			if(SSftl_navigation.current_sector.planet) dat += "\A [SSftl_navigation.current_sector.planet] is within range of the FOB.<br>"
+			if(SSftl_navigation.current_sector.station) dat += "\A [SSftl_navigation.current_sector.station] is within trading range.<br>"
```
Comment (ninjanomnom) [inline ftl/code/machinery/computer/navigation_computer.dm]: Same as above

File: ftl/code/machinery/computer/navigation_computer.dm [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+			dat += "<hr>Adjacent systems:<hr>"
+			for(var/datum/sector/S in SSftl_navigation.current_sector.connected_sectors)
+				dat += "<A href='?src=[REF(src)];details=[S.id]'>[S.name]</A><br>"
+
+		if("Details")
+			dat += "<A href='?src=[REF(src)];options=1'>Return to current system</A><br><hr>"
+			dat += "Long range scan of [sector_details.name]:<br>"
+			//long range results
+			if(sector_details.planet) dat += "Star charts indicate a planet will be in this region.<br>"
+			if(sector_details.station) dat += "The Galactic Trading Association lists an operational station open for trading here.<br>"
```
Comment (ninjanomnom) [inline ftl/code/machinery/computer/navigation_computer.dm]: Same as above

File: ftl/code/turfs/ftl_space.dm [hunk 27]
```diff
@@ -0,0 +1,58 @@
...
+		icon_state = SPACE_ICON_STATE //Update us using the normal space define
+
+
+/turf/open/space/transit/ftl/throw_atom(atom/movable/AM) //TODO: Talk about how we handle players who fall off the ship
+	set waitfor = FALSE
+	if(!AM || istype(AM, /obj/docking_port))
+		return
+	if(AM.loc != src) 	// Multi-tile objects are "in" multiple locs but its loc is it's true placement.
+		return			// Don't move multi tile objects if their origin isnt in transit
+	if(SSftl_navigation.ftl_state != FTL_JUMPING) return //Only throw if we are in transit
```
Comment (ninjanomnom) [inline ftl/code/turfs/ftl_space.dm]: Same as above

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 13 above)
Comment (TMTIME) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: This STB loops through the ship's z level replacing normal space turfs with our custom transit ones.

If it detects any it sends a message during init that it found X tiles and took Y seconds to fix, like how the atmos subsystem reports any roundstart active atmos turfs

Comment (ninjanomnom) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: No that's what the loop is doing, why is this using sub turf blocks instead of a simple
`for(var/turf/place in block(locate(1,1,z), locate(255,255,z)))`

File: ftl/code/controllers/subsystem/ftl_navigation.dm [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+		var/datum/sector/S = sector_typeless
+		generate_connecting_sectors(S)
+
+	var/timer = world.timeofday
+	var/count = 0
+	var/msg
+	for(var/z in SSmapping.levels_by_trait(ZTRAIT_STATION)) //Temp fix to make any map work with ftl transit states
+		for(var/turf/T in block(locate(1,1,z), locate(255,255,z)))
+			if(T.type == /turf/open/space || T.type == /turf/open/space/basic) //I want types only, not subtypes too
+				T.ChangeTurf(/turf/open/space/transit/ftl)
```
Comment (ninjanomnom) [inline ftl/code/controllers/subsystem/ftl_navigation.dm]: As I said before you should load the ship as a template onto a blank z level of transit space in the first place. Then you just have noop in the template

**Comment by TMTIME** [comment]
^ that commit should fail if I got unit tests working right

File: ftl/code/modules/unit_tests/ship_template.dm [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+    var/datum/parsed_map/P = new(initial(spawner.template_name)) //Parse the template
+    P.build_cache() //Parse grid_models
+    var/list/count = list()
+    for(var/model in P.modelCache)
+      for(var/T in P.modelCache[model][1])
+        if((T.type == /turf/open/space || T.type == /turf/open/space/basic) //Replace only these space turfs)
+          count[T] += 1
+    if(count.len > 0)
+      for(var/turf in count)
+        failreason += "[count[turf]] [turf] exist within [initial(spawner.template_name)]. Replace with '/turf/open/space/transit/ftl' | "
```
Comment (ninjanomnom) [inline ftl/code/modules/unit_tests/ship_template.dm]: The map you load the template on should be made of transit/ftl and the template should use noop turfs. That way you can use the baseturf system so that underneath the template is all transit/ftl

Comment (TMTIME) [reply ftl/code/modules/unit_tests/ship_template.dm]: That's how it currently works. This is just the unit test to fail travis but it's not working rn

**Comment by TMTIME** [comment]
Also it looks like Travis didn't init the nav subsystem during its checks. https://travis-ci.org/FTL13/New-FTL13/jobs/435270751#L1283

**Comment by TMTIME** [comment]
Same again + the unit_test didn't even start

File: ftl/code/modules/unit_tests/ship_template.dm (hunk 29 above)
Comment (ninjanomnom) [reply ftl/code/modules/unit_tests/ship_template.dm]: The message says to replace with `/turf/open/space/transit/ftl` though when it should be saying to replace with noop

**Comment by TMTIME** [comment]
checkem 
https://travis-ci.org/FTL13/New-FTL13/jobs/435797671#L1469

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 1 above)
Comment (Dennok) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: This change block possibility of random frequency in New(). I dont know, but random frequency make uneverse more randomly and interresting,,, may be.

Comment (Qustinnus) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: @Dennok If you wanted to have random frequency you could just do it in this for loop

**Comment by TMTIME** [comment]
This is good to review again btw

**Comment by Dennok** [comment]
Just lost here comment to have it in participating.

File: ftl/code/controllers/subsystem/ftl_navigation.dm (hunk 15 above)
Comment (ninjanomnom) [reply ftl/code/controllers/subsystem/ftl_navigation.dm]: You still haven't addressed this
